internal/addr_test.go

```diff
@@ -40,6 +40,10 @@ func TestParseAddr(t *testing.T) {
 		{Addr: "foo", Target: parseAddrTarget{Host: "foo", Port: 0, Err: nil}},
 		{Addr: ":1234", Target: parseAddrTarget{Host: "127.0.0.1", Port: 1234, Err: nil}},
 		{Addr: "", Target: parseAddrTarget{Host: "127.0.0.1", Port: 0, Err: nil}},
+		{Addr: "169.254.172.39", Target: parseAddrTarget{Host: "169.254.172.39", Port: 0, Err: nil}},
+		{Addr: "169.254.172.39:5703", Target: parseAddrTarget{Host: "169.254.172.39", Port: 5703, Err: nil}},
+		{Addr: "fe80::43:ecff:fec9:1683", Target: parseAddrTarget{Host: "fe80::43:ecff:fec9:1683", Port: 0, Err: nil}},
+		{Addr: "[fe80::43:ecff:fec9:1683]:5705", Target: parseAddrTarget{Host: "fe80::43:ecff:fec9:1683", Port: 5705, Err: nil}},
```

Comment:
@semihbkgr Could you also add a test case for `[fe80::43:ecff:fec9:1683` and `fe80::43:ecff:fec9:1683]:5678`?

Reply:
Ok, we assume that these are not valid addresses, right?